### PR TITLE
[DAG] Apply Disjoint flag.

### PR DESCRIPTION
**Pull request by fengfeng09**
or disjoint (or disjoint (x, c0), c1)
-->
or disjont x, or (c0, c1)
Alive2: https://alive2.llvm.org/ce/z/3wPth5

### CODE REVIEW

**Review by dtcxzyw — COMMENTED** [review]
Can you add some tests to demonstrate that this patch improves the codegen?

**Comment by fengfeng09** [comment]
> Can you add some tests to demonstrate that this patch improves the codegen?

Ok.

**Comment by fengfeng09** [comment]
> Can you add some tests to demonstrate that this patch improves the codegen?

Updated. A test case is appended.

**Review by dtcxzyw — APPROVED** [review]
LGTM.

**Comment by fengfeng09** [comment]
@dtcxzyw Thanks, could you help me merge this PR? I have no write access to this repo.

File: llvm/test/CodeGen/AArch64/apply-disjoint-flag-in-dagcombine.ll [hunk 1]
```diff
@@ -0,0 +1,15 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=aarch64-none-eabi -verify-machineinstrs %s  -start-before=aarch64-isel -o - | FileCheck %s
```
Comment (nikic) [inline llvm/test/CodeGen/AArch64/apply-disjoint-flag-in-dagcombine.ll:2]: Why does this need `-start-before=aarch64-isel`? (Or `-verify-machineinstrs` for that matter)

Comment (fengfeng09) [reply llvm/test/CodeGen/AArch64/apply-disjoint-flag-in-dagcombine.ll]: Cleared. This is not really needed.

**Review by arsenm — APPROVED** [review]

**Pull request merged**